database_interaction/database_select.py

```
@@ -1,9 +1,65 @@
+from datetime import datetime
+from database_interaction import database
+from filters import (
+    name_filter,
+    author_filter,
+    type_filter,
+    tag_filter,
+    rating_filter,
+    time_filter,
+)
 from database_interaction import database
 
 
 def select(id):
-    song = database.db.session.query(
-        database.song).filter_by(id=id).first_or_404()
+    song = database.db.session.query(database.song).filter_by(id=id).first_or_404()
     model_dict = vars(song)
-    del model_dict['_sa_instance_state']
+    del model_dict["_sa_instance_state"]
     return model_dict
+
+
+def select_by_filters(st, page):
+    my_obj_dict = {}
+    my_list = []
+    my_select_dict = database.db.session.query(database.song)
+    for i in st:
+        if i == "name":
+            name = st.get("name")
+            my_select_dict = name_filter.filter_by_name(name, my_select_dict)
+        if i == "author":
+            author = st.get("author")
+            my_select_dict = author_filter.filter_by_author(author, my_select_dict)
+        if i == "type":
+            type = st.get("type")
+            my_select_dict = type_filter.filter_by_type(type, my_select_dict)
+        if i == "tag":
+            tag = st.get("tag")
+            my_select_dict = tag_filter.filter_by_tag(tag, my_select_dict)
+        if i == "rating":
+            rating = float(st.get("rating"))
+            my_select_dict = rating_filter.filter_by_rating(rating, my_select_dict)
+        if i == "time_created":
+            try:
+                time_created = datetime.strptime(st.get("time_created"), "%m/%d/%Y")
+            except ValueError:
+                return False
```

throw exception instead of returning true or false